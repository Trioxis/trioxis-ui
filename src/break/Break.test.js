import React from 'react';
import { mount } from 'enzyme';
import {Break} from './';

jest.mock('react-jss',()=>{
  jest.spyOn(global.Math,'random').mockImplementation(()=>0.42);
  return jest.requireActual('react-jss')
})

test('Break should render as snapshot defines', () => {
  const tree = mount(<Break />)
  expect(tree).toMatchSnapshot()
});
