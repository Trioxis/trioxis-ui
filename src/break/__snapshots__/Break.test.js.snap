// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Break should render as snapshot defines 1`] = `
<Jss(_Break)>
  <_Break
    classes={
      Object {
        "content": "content-0-3 content-0-1",
        "wrapper": "wrapper-0-2 wrapper-0-0",
      }
    }
    sheet={
      StyleSheet {
        "attached": true,
        "classes": Object {
          "content": "content-0-3 content-0-1",
          "wrapper": "wrapper-0-2 wrapper-0-0",
        },
        "deployed": true,
        "linked": true,
        "options": Object {
          "Renderer": [Function],
          "classes": Object {
            "content": "content-0-3 content-0-1",
            "wrapper": "wrapper-0-2 wrapper-0-0",
          },
          "generateClassName": [Function],
          "index": -99998,
          "insertionPoint": undefined,
          "jss": Jss {
            "generateClassName": [Function],
            "options": Object {
              "Renderer": [Function],
              "createGenerateClassName": [Function],
              "plugins": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
            },
            "plugins": PluginsRegistry {
              "hooks": Object {
                "onChangeValue": Array [
                  [Function],
                  [Function],
                ],
                "onCreateRule": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "onProcessRule": Array [
                  [Function],
                  [Function],
                ],
                "onProcessSheet": Array [],
                "onProcessStyle": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
              },
            },
            "version": "8.1.0",
          },
          "link": true,
          "meta": "Jss(_Break), Unthemed, Dynamic",
          "parent": [Circular],
          "sheet": [Circular],
        },
        "renderer": DomRenderer {
          "element": <style
            data-jss=""
            data-meta="Jss(_Break), Unthemed, Dynamic"
            type="text/css"
          >
            
.wrapper-0-2 {
  background: false;
}
.content-0-3 {
}

          </style>,
          "getSelector": [Function],
          "getStyle": [Function],
          "hasInsertedRules": false,
          "setSelector": [Function],
          "setStyle": [Function],
          "sheet": [Circular],
        },
        "rules": RuleList {
          "classes": Object {
            "content": "content-0-3 content-0-1",
            "wrapper": "wrapper-0-2 wrapper-0-0",
          },
          "index": Array [
            Object {
              "background": false,
            },
            Object {
              "color": undefined,
            },
          ],
          "map": Object {
            ".content-0-3": Object {
              "color": undefined,
            },
            ".wrapper-0-2": Object {
              "background": false,
            },
            "content": Object {
              "color": undefined,
            },
            "wrapper": Object {
              "background": false,
            },
          },
          "options": Object {
            "Renderer": [Function],
            "classes": Object {
              "content": "content-0-3 content-0-1",
              "wrapper": "wrapper-0-2 wrapper-0-0",
            },
            "generateClassName": [Function],
            "index": -99998,
            "insertionPoint": undefined,
            "jss": Jss {
              "generateClassName": [Function],
              "options": Object {
                "Renderer": [Function],
                "createGenerateClassName": [Function],
                "plugins": Array [
                  Object {
                    "onCreateRule": [Function],
                    "onProcessRule": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                ],
              },
              "plugins": PluginsRegistry {
                "hooks": Object {
                  "onChangeValue": Array [
                    [Function],
                    [Function],
                  ],
                  "onCreateRule": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onProcessRule": Array [
                    [Function],
                    [Function],
                  ],
                  "onProcessSheet": Array [],
                  "onProcessStyle": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
              },
              "version": "8.1.0",
            },
            "link": true,
            "meta": "Jss(_Break), Unthemed, Dynamic",
            "parent": [Circular],
            "sheet": [Circular],
          },
          "raw": Object {
            "content": Object {
              "color": [Function],
              "composes": "content-0-1",
            },
            "wrapper": Object {
              "background": [Function],
              "composes": "wrapper-0-0",
            },
          },
        },
      }
    }
  >
    <div
      className={undefined}
    >
      <Jss(_Section)
        className="wrapper-0-2 wrapper-0-0"
      >
        <_Section
          className="wrapper-0-2 wrapper-0-0"
          classes={
            Object {
              "wrapper": "wrapper-0-4",
            }
          }
          sheet={
            StyleSheet {
              "0.42": null,
              "attached": true,
              "classes": Object {
                "wrapper": "wrapper-0-4",
              },
              "deployed": true,
              "linked": false,
              "options": Object {
                "Renderer": [Function],
                "classes": Object {
                  "wrapper": "wrapper-0-4",
                },
                "generateClassName": [Function],
                "index": -99999,
                "insertionPoint": undefined,
                "jss": Jss {
                  "generateClassName": [Function],
                  "options": Object {
                    "Renderer": [Function],
                    "createGenerateClassName": [Function],
                    "plugins": Array [
                      Object {
                        "onCreateRule": [Function],
                        "onProcessRule": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessRule": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                    ],
                  },
                  "plugins": PluginsRegistry {
                    "hooks": Object {
                      "onChangeValue": Array [
                        [Function],
                        [Function],
                      ],
                      "onCreateRule": Array [
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                      ],
                      "onProcessRule": Array [
                        [Function],
                        [Function],
                      ],
                      "onProcessSheet": Array [],
                      "onProcessStyle": Array [
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                      ],
                    },
                  },
                  "version": "8.1.0",
                },
                "meta": "Jss(_Section), Unthemed, Static",
                "parent": [Circular],
                "sheet": [Circular],
              },
              "renderer": DomRenderer {
                "element": <style
                  data-jss=""
                  data-meta="Jss(_Section), Unthemed, Static"
                  type="text/css"
                >
                  
.wrapper-0-4 {
  width: 100%;
  padding-bottom: 3em;
  background-color: #FBFCFC;
}

                </style>,
                "getSelector": [Function],
                "getStyle": [Function],
                "hasInsertedRules": false,
                "setSelector": [Function],
                "setStyle": [Function],
                "sheet": [Circular],
              },
              "rules": RuleList {
                "classes": Object {
                  "wrapper": "wrapper-0-4",
                },
                "index": Array [
                  Object {
                    "background-color": "#FBFCFC",
                    "padding-bottom": "3em",
                    "width": "100%",
                  },
                ],
                "map": Object {
                  ".wrapper-0-4": Object {
                    "background-color": "#FBFCFC",
                    "padding-bottom": "3em",
                    "width": "100%",
                  },
                  "wrapper": Object {
                    "background-color": "#FBFCFC",
                    "padding-bottom": "3em",
                    "width": "100%",
                  },
                },
                "options": Object {
                  "Renderer": [Function],
                  "classes": Object {
                    "wrapper": "wrapper-0-4",
                  },
                  "generateClassName": [Function],
                  "index": -99999,
                  "insertionPoint": undefined,
                  "jss": Jss {
                    "generateClassName": [Function],
                    "options": Object {
                      "Renderer": [Function],
                      "createGenerateClassName": [Function],
                      "plugins": Array [
                        Object {
                          "onCreateRule": [Function],
                          "onProcessRule": [Function],
                        },
                        Object {
                          "onProcessStyle": [Function],
                        },
                        Object {
                          "onProcessStyle": [Function],
                        },
                        Object {
                          "onProcessStyle": [Function],
                        },
                        Object {
                          "onProcessStyle": [Function],
                        },
                        Object {
                          "onChangeValue": [Function],
                          "onProcessStyle": [Function],
                        },
                        Object {
                          "onProcessStyle": [Function],
                        },
                        Object {
                          "onChangeValue": [Function],
                          "onProcessRule": [Function],
                          "onProcessStyle": [Function],
                        },
                        Object {
                          "onProcessStyle": [Function],
                        },
                      ],
                    },
                    "plugins": PluginsRegistry {
                      "hooks": Object {
                        "onChangeValue": Array [
                          [Function],
                          [Function],
                        ],
                        "onCreateRule": Array [
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                        ],
                        "onProcessRule": Array [
                          [Function],
                          [Function],
                        ],
                        "onProcessSheet": Array [],
                        "onProcessStyle": Array [
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                          [Function],
                        ],
                      },
                    },
                    "version": "8.1.0",
                  },
                  "meta": "Jss(_Section), Unthemed, Static",
                  "parent": [Circular],
                  "sheet": [Circular],
                },
                "raw": Object {
                  "wrapper": Object {
                    "backgroundColor": "#FBFCFC",
                    "paddingBottom": "3em",
                    "width": "100%",
                  },
                },
              },
            }
          }
        >
          <div
            className="wrapper-0-4 wrapper-0-2 wrapper-0-0"
            style={undefined}
          >
            <Jss(_ContentContainer)>
              <_ContentContainer
                classes={
                  Object {
                    "app": "app-0-5",
                  }
                }
                sheet={
                  StyleSheet {
                    "0.42": null,
                    "attached": true,
                    "classes": Object {
                      "app": "app-0-5",
                    },
                    "deployed": true,
                    "linked": false,
                    "options": Object {
                      "Renderer": [Function],
                      "classes": Object {
                        "app": "app-0-5",
                      },
                      "generateClassName": [Function],
                      "index": -100000,
                      "insertionPoint": undefined,
                      "jss": Jss {
                        "generateClassName": [Function],
                        "options": Object {
                          "Renderer": [Function],
                          "createGenerateClassName": [Function],
                          "plugins": Array [
                            Object {
                              "onCreateRule": [Function],
                              "onProcessRule": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onChangeValue": [Function],
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onChangeValue": [Function],
                              "onProcessRule": [Function],
                              "onProcessStyle": [Function],
                            },
                            Object {
                              "onProcessStyle": [Function],
                            },
                          ],
                        },
                        "plugins": PluginsRegistry {
                          "hooks": Object {
                            "onChangeValue": Array [
                              [Function],
                              [Function],
                            ],
                            "onCreateRule": Array [
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                            ],
                            "onProcessRule": Array [
                              [Function],
                              [Function],
                            ],
                            "onProcessSheet": Array [],
                            "onProcessStyle": Array [
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                            ],
                          },
                        },
                        "version": "8.1.0",
                      },
                      "meta": "Jss(_ContentContainer), Unthemed, Static",
                      "parent": [Circular],
                      "sheet": [Circular],
                    },
                    "renderer": DomRenderer {
                      "element": <style
                        data-jss=""
                        data-meta="Jss(_ContentContainer), Unthemed, Static"
                        type="text/css"
                      >
                        
.app-0-5 {
  margin: auto;
  max-width: 1200px;
}

                      </style>,
                      "getSelector": [Function],
                      "getStyle": [Function],
                      "hasInsertedRules": false,
                      "setSelector": [Function],
                      "setStyle": [Function],
                      "sheet": [Circular],
                    },
                    "rules": RuleList {
                      "classes": Object {
                        "app": "app-0-5",
                      },
                      "index": Array [
                        Object {
                          "margin": "auto",
                          "max-width": "1200px",
                        },
                      ],
                      "map": Object {
                        ".app-0-5": Object {
                          "margin": "auto",
                          "max-width": "1200px",
                        },
                        "app": Object {
                          "margin": "auto",
                          "max-width": "1200px",
                        },
                      },
                      "options": Object {
                        "Renderer": [Function],
                        "classes": Object {
                          "app": "app-0-5",
                        },
                        "generateClassName": [Function],
                        "index": -100000,
                        "insertionPoint": undefined,
                        "jss": Jss {
                          "generateClassName": [Function],
                          "options": Object {
                            "Renderer": [Function],
                            "createGenerateClassName": [Function],
                            "plugins": Array [
                              Object {
                                "onCreateRule": [Function],
                                "onProcessRule": [Function],
                              },
                              Object {
                                "onProcessStyle": [Function],
                              },
                              Object {
                                "onProcessStyle": [Function],
                              },
                              Object {
                                "onProcessStyle": [Function],
                              },
                              Object {
                                "onProcessStyle": [Function],
                              },
                              Object {
                                "onChangeValue": [Function],
                                "onProcessStyle": [Function],
                              },
                              Object {
                                "onProcessStyle": [Function],
                              },
                              Object {
                                "onChangeValue": [Function],
                                "onProcessRule": [Function],
                                "onProcessStyle": [Function],
                              },
                              Object {
                                "onProcessStyle": [Function],
                              },
                            ],
                          },
                          "plugins": PluginsRegistry {
                            "hooks": Object {
                              "onChangeValue": Array [
                                [Function],
                                [Function],
                              ],
                              "onCreateRule": Array [
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                              ],
                              "onProcessRule": Array [
                                [Function],
                                [Function],
                              ],
                              "onProcessSheet": Array [],
                              "onProcessStyle": Array [
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                                [Function],
                              ],
                            },
                          },
                          "version": "8.1.0",
                        },
                        "meta": "Jss(_ContentContainer), Unthemed, Static",
                        "parent": [Circular],
                        "sheet": [Circular],
                      },
                      "raw": Object {
                        "app": Object {
                          "margin": "auto",
                          "maxWidth": "1200px",
                        },
                      },
                    },
                  }
                }
              >
                <div
                  className="app-0-5"
                >
                  <div
                    className="content-0-3 content-0-1"
                  />
                </div>
              </_ContentContainer>
            </Jss(_ContentContainer)>
          </div>
        </_Section>
      </Jss(_Section)>
    </div>
  </_Break>
</Jss(_Break)>
`;
